packages/cfd/src/Components/__tests__/cfd-poa.spec.js

```diff
@@ -0,0 +1,296 @@
+
+    it('should render the POA component', async () => {
+        render(<CFDPOA {...props} />);
+        await waitFor(() => {
+            expect(screen.getByText(/Address information/i)).toBeInTheDocument();
+        });
+    });
+
+    it('should render the POA component with the correct texts and buttons', async () => {
+        render(<CFDPOA {...props} />);
+        await waitFor(() => {
+            expect(screen.getByText(/Address information/i)).toBeInTheDocument();
+            expect(screen.getByText(/First line of address*/i)).toBeInTheDocument();
+            expect(screen.getByText(/Second line of address \(optional\)/i)).toBeInTheDocument();
+            expect(screen.getByText('Town/City*')).toBeInTheDocument();
+            expect(screen.getByText('State/Province')).toBeInTheDocument();
+            expect(screen.getByText('Postal/ZIP code')).toBeInTheDocument();
+            expect(screen.getByText(/FileUploaderContainer/i)).toBeInTheDocument();
+            expect(screen.getByRole('button', { name: /Previous/i })).toBeInTheDocument();
+            expect(screen.getByRole('button', { name: /Submit/i })).toBeInTheDocument();
+        });
+    });
+    it('should render the correct input for all the fields', async () => {
+        render(<CFDPOA {...props} onSubmit={() => true} />);
+
+        await waitFor(() => {
+            expect(screen.getByText(/Address information/i)).toBeInTheDocument();
+        });
+
+        const address_line_1_input = screen.getByPlaceholderText(address_line_1);
+        const first_line_address_text = 'Test first line address';
+        expect(address_line_1_input.value).toBe('');
+        fireEvent.change(address_line_1_input, { target: { value: first_line_address_text } });
+        await waitFor(() => {
+            expect(address_line_1_input.value).toBe(first_line_address_text);
+        });
+
+        const address_line_2_input = screen.getByPlaceholderText(address_line_2);
+        const second_line_address_text = 'Test second line address';
+        expect(address_line_2_input.value).toBe('');
+        fireEvent.change(address_line_2_input, { target: { value: second_line_address_text } });
+        await waitFor(() => {
+            expect(address_line_2_input.value).toBe(second_line_address_text);
+        });
+
+        const address_town_input = screen.getByPlaceholderText(address_town);
+        const town_text = 'Test town';
+        expect(address_town_input.value).toBe('');
+        fireEvent.change(address_town_input, { target: { value: town_text } });
+        await waitFor(() => {
+            expect(address_town_input.value).toBe(town_text);
+        });
+
+        const address_state_input = screen.getByPlaceholderText(address_state);
+        const address_state_text = 'Test state';
+        expect(address_state_input.value).toBe('Default test state');
+        fireEvent.change(address_state_input, { target: { value: address_state_text } });
+        await waitFor(() => {
```

Comment:
Isn't it possible to use one `await waitFor` for all these `expect` lines?

Reply:
I thought it will be more organized for each text fields

Reply:
But this leads to unnecessary code duplication
waitFor is called many times yet all the actions performed can be done inside one waitFor.